### PR TITLE
fix: #15726 increase max recurring events limit input

### LINKED ISSUE #15726: Maximimum events for recurring event has been reduced - can no longer book more than 20
Found a bug? Please fill out the sections below. 👍

### Issue Summary

I sell coaching packages. 24 sessions either weekly or fortnightly. I have a 24 weekly session package that works fine. I had a 24 fortnightly package that I didn't realise was set to a maximum of 12, so I needed to fix it to be 24. When I edit it I get the error "variable must be less than or equal to 20". But I already have one that is 24! So someone must have changed this variable limit. 

I tried to duplicate my existing 24 weekly session booking but it wouldn't let me save it when I changed weekly to fortnightly. 

### Steps to Reproduce

1. (for example) Went to ... "recurring" inside the event
2. Clicked on... the number of appointments to book, changed it to 24 and clicked save.
3. ...

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

Because I already have an event with more than 20 sessions that is currently active. 

### Actual Results

- What's happening right now that is different from what is expected

### Expected Results

- This is an ideal result that the system should get after the tests are performed

### Technical details

- Browser version, screen recording, console logs, network requests: You can make a recording with [Bird Eats Bug](https://birdeatsbug.com/). (Chrome latest version
25d5d1a6-cf02-4414-8fd5-d0d73a68b824

, MacOS Sonoma 14.1.1)
- Node.js version (don't know - sorry I'm not a developer)
- Anything else that you think could be an issue.

### Evidence

- How was this tested? This is quite mandatory in terms of bugs. Providing evidence of your testing with screenshots or/and videos is an amazing way to prove the bug and a troubleshooting chance to find the solution.

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

- Fixes #15726 
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by sean-brydon** [comment]
@PeerRich @keithwillcode is this something we plan to do - There was a very good reason for limiting this to 20 im not sure if we plan to increase it while this œuery is still on slow side

**Review by mahir22B — APPROVED** [review]

**Comment by keithwillcode** [comment]
Yeah, we had reduced it actually because of the difficulties of booking so far out. @anikdhabal Did you test this with 40? If there isn't a performance issue, we can try and go ahead with this change as I recall we changed how booking recurring events works so it's more flexible in working even when it runs into conflicts. 

@CarinaWolli You may have thoughts here as well.

**Comment by CarinaWolli** [comment]
> Yeah, we had reduced it actually because of the difficulties of booking so far out. @anikdhabal Did you test this with 40? If there isn't a performance issue, we can try and go ahead with this change as I recall we changed how booking recurring events works so it's more flexible in working even when it runs into conflicts.
> 
> @CarinaWolli You may have thoughts here as well.

As it is now, it shouldn't be an issue performance-wise because we only check the availability for the first two slots. I am just wondering if this was a temporary solution to fix recurring events and if we may plan in the future to again check all slots for their availability along with other performance improvements.

Also not related to this question, but we don't mention this anywhere for the user that only the first two slots are checked for availability (cc @ciaranha): 
Screenshot 2024-07-23 at 7 55 41 AM

**Comment by Amit91848** [comment]
> As it is now, it shouldn't be an issue performance-wise because we only check the availability for the first two slots. I am just wondering if this was a temporary solution to fix recurring events and if we may plan in the future to again check all slots for their availability along with other performance improvements.
> 
> Also not related to this question, but we don't mention this anywhere for the user that only the first two slots are checked for availability (cc @ciaranha)

This has been reported by others #15997

**Review by CarinaWolli — CHANGES_REQUESTED** [review]

File: apps/web/components/eventtype/RecurringEventController.tsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ export default function RecurringEventController({
                         disabled={recurringLocked.disabled}
                         type="number"
                         min="1"
-                        max="20"
+                        max="40"
```
[the remark this answers is not in the document]
Comment (vijayraghav-io) [reply apps/web/components/eventtype/RecurringEventController.tsx:121]: sure, will update.

[the remark this answers is not in the document]
Comment (vijayraghav-io) [reply apps/web/components/eventtype/RecurringEventController.tsx]: done

**Review by CarinaWolli — APPROVED** [review]
Thank you 🙏

**Pull request merged**